websites/M/MyTF1/presence.ts

```diff
@@ -0,0 +1,165 @@
+const presence = new Presence({
+		clientId: "1043951049800695808",
+	}),
+	strings = presence.getStrings({
+		viewHome: "general.viewHome",
+		viewPage: "general.viewPage",
+		viewProfile: "general.viewProfile",
+		watchingVid: "general.watchingVid",
+		browse: "general.browsing",
+		live: "general.live",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+const enum Assets {
+	myTf1 = "https://i.imgur.com/k8jIoSZ.png",
+	tf1 = "https://i.imgur.com/pFgnoUS.png",
+	tfx = "https://i.imgur.com/u5dTkHx.png",
+	tmc = "https://i.imgur.com/62uYw2r.png",
+	lci = "https://i.imgur.com/qrRgNgk.png",
+	tf1Series = "https://i.imgur.com/WOmFIm3.png",
+	live = "https://i.imgur.com/xuvR04J.png",
+	replay = "https://i.imgur.com/OZDAx9n.png",
+}
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: Assets.myTf1,
+			startTimestamp: browsingTimestamp,
+		},
+		path = document.location.pathname,
+		time = await presence.getSetting<boolean>("time");
+
+	if (!time) delete presenceData.startTimestamp;
+
+	const { browse, live } = await strings;
+
+	if (path === "/") {
+		presenceData.details = (await strings).viewHome;
+		presenceData.state = browse;
+	} else if (path.startsWith("/tf1/direct")) {
+		presenceData.details = document.querySelector(
+			"[class*=VideoSummary_programName_xXfYS]"
+		).textContent;
+		presenceData.state = document.querySelector(
+			"[class*=VideoSummary_title_o8ZzQ]"
+		).textContent;
+		presenceData.smallImageKey = Assets.live;
+		presenceData.smallImageText = live;
+		if (
+			document.querySelector<HTMLImageElement>(
+				"[class*=VideoSummary_programLink_xY8sS] picture img"
+			).src
+		) {
+			presenceData.largeImageKey = document.querySelector<HTMLImageElement>(
+				"[class*=VideoSummary_programLink_xY8sS] picture img"
+			).src;
```

Comment:
Note: **Applies to multiple instances.**

For better future support, remove the 'computer generated' part of the selectors. If these don't work, let me know.
```diff
- "[class*=Human_Readable_Thing_Shfa52d]"
+ "[class*=Human_Readable_Thing]"
```